Index: /opt/spm12/@file_array/private/file2mat.c
===================================================================
--- /opt/spm12/@file_array/private/file2mat.c	(revision 7468)
+++ /opt/spm12/@file_array/private/file2mat.c	(working copy)
@@ -52 +52 @@
-int page_size()
+static int page_size()
@@ -92 +92 @@
-void get_1(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned char idat[],
+static void get_1(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned char idat[],
@@ -98 +98 @@
-void get_8(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned char idat[],
+static void get_8(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned char idat[],
@@ -115 +115 @@
-void get_16(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned short idat[],
+static void get_16(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned short idat[],
@@ -132 +132 @@
-void get_32(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned int idat[],
+static void get_32(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned int idat[],
@@ -149 +149 @@
-void get_64(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned long long idat[],
+static void get_64(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned long long idat[],
@@ -166 +166 @@
-void get_w8(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned char idat[],
+static void get_w8(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned char idat[],
@@ -188 +188 @@
-void get_w16(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned short idat[],
+static void get_w16(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned short idat[],
@@ -210 +210 @@
-void get_w32(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned int idat[],
+static void get_w32(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned int idat[],
@@ -232 +232 @@
-void get_w64(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned long long idat[],
+static void get_w64(mwSize ndim, mwSize idim[], unsigned long long *iptr[], unsigned long long idat[],
@@ -254 +254 @@
-void swap8(long long n, unsigned char d[])
+static void swap8(long long n, unsigned char d[])
@@ -257 +257 @@
-void swap16(long long n, unsigned char d[])
+static void swap16(long long n, unsigned char d[])
@@ -266 +266 @@
-void swap32(long long n, unsigned char d[])
+static void swap32(long long n, unsigned char d[])
@@ -276 +276 @@
-void swap64(long long n, unsigned char d[])
+static void swap64(long long n, unsigned char d[])
@@ -297 +297 @@
-Dtype table[] = {
+static Dtype table[] = {
@@ -327 +327 @@
-void werror(char *where, DWORD last_error)
+static void werror(char *where, DWORD last_error)
@@ -345 +345 @@
-void werror(char *where, int last_error)
+static void werror(char *where, int last_error)
@@ -355 +355 @@
-void do_unmap_file(MTYPE *map)
+static void do_unmap_file(MTYPE *map)
@@ -373 +373 @@
-const double *getpr(const mxArray *ptr, const char nam[], int len, int *n)
+static const double *getpr(const mxArray *ptr, const char nam[], int len, int *n)
@@ -415 +415 @@
-void do_map_file(const mxArray *ptr, MTYPE *map)
+static void do_map_file(const mxArray *ptr, MTYPE *map)
Index: /opt/spm12/@file_array/private/mat2file.c
===================================================================
--- /opt/spm12/@file_array/private/mat2file.c	(revision 7468)
+++ /opt/spm12/@file_array/private/mat2file.c	(working copy)
@@ -44 +44 @@
-void swap8(int n, unsigned char id[], unsigned char od[])
+static void swap8(int n, unsigned char id[], unsigned char od[])
@@ -53 +53 @@
-void swap16(int n, unsigned char id[], unsigned char od[])
+static void swap16(int n, unsigned char id[], unsigned char od[])
@@ -62 +62 @@
-void swap32(int n, unsigned char id[], unsigned char od[])
+static void swap32(int n, unsigned char id[], unsigned char od[])
@@ -72 +72 @@
-void swap64(int n, unsigned char id[], unsigned char od[])
+static void swap64(int n, unsigned char id[], unsigned char od[])
@@ -85 +85 @@
-Dtype table[] = {
+static Dtype table[] = {
@@ -108,3 +108,3 @@
-off_t icumprod[MXDIMS], ocumprod[MXDIMS];
-off_t poff;
-long len;
+static off_t icumprod[MXDIMS], ocumprod[MXDIMS];
+static off_t poff;
+static long len;
@@ -112 +112 @@
-unsigned char wbuf[BLEN], *dptr;
+static unsigned char wbuf[BLEN], *dptr;
@@ -114 +114 @@
-void put_bytes(int ndim, FILE *fp, int *ptr[], int idim[], unsigned char idat[], off_t indo, off_t indi, void (*swap)())
+static void put_bytes(int ndim, FILE *fp, int *ptr[], int idim[], unsigned char idat[], off_t indo, off_t indi, void (*swap)())
@@ -157 +157 @@
-void put(FTYPE map, int *ptr[], int idim[], void *idat)
+static void put(FTYPE map, int *ptr[], int idim[], void *idat)
@@ -190 +190 @@
-const double *getpr(const mxArray *ptr, const char nam[], int len, int *n)
+static const double *getpr(const mxArray *ptr, const char nam[], int len, int *n)
@@ -233 +233 @@
-void open_file(const mxArray *ptr, FTYPE *map)
+static void open_file(const mxArray *ptr, FTYPE *map)
@@ -302 +302 @@
-void close_file(FTYPE map)
+static void close_file(FTYPE map)
